json.c

```diff
@@ -343,25 +344,42 @@ static int json_get_string_size(struct json_parse_state_s *state,
         //      4       21      U + 10000       U + 10FFFF      11110xxx        10xxxxxx        10xxxxxx        10xxxxxx
         // Note: the high and low surrogate halves used by UTF-16 (U+D800 through U+DFFF) and code points not encodable by UTF-16 (those after U+10FFFF) are not legal Unicode values, and their UTF-8 encoding must be treated as an invalid byte sequence.
     
-        if (codepoint <= 0x7f) {
-          offset += 1;
-          data_size += 1;
+        if (high_surrogate_codepoint) {
```

Comment:
I don't really know what a `high_surrogate_codepoint` is - could you add some comments?